### DISCUSSION TITLE
Key scrolling

### DISCUSSION THREAD

**Discussion by skearya** [Hammerspoon/hammerspoon#3255] [Q&A] [answered]
Hello,
I want to have the up key scroll just a bit up, and the same thing for the down key. I looked into this for a while but did not find any answers, help appreciated.

**Comment by MuhammedZakir** [comment, marked as answer]
You can do that by creating hotkeys [1] for arrow keys that will post [2] scroll events [3] when triggered.

[1] http://www.hammerspoon.org/docs/hs.hotkey.html#new
[2] http://www.hammerspoon.org/docs/hs.eventtap.event.html#post
[3] http://www.hammerspoon.org/docs/hs.eventtap.event.html#newScrollEvent

**Reply by skearya** [reply]
Thank you for the help, could you write the script for one of the keys? I don't have lua experience

**Reply by MuhammedZakir** [reply]
Try (I didn't test this):

```lua
local hotkey = require('hs.hotkey')
local eventtap_event = require('hs.eventtap.event')

local function scrollVertically(y)
  eventtap_event.newScrollEvent({0, y}, {}):post()
end

arrowUp2ScrollUp = hotkey.new("", "up",
  -- when pressed
  function() scrollVertically(3) end,
  -- when repeating / while held down
  function() scrollVertically(1) end
):enable()

arrowDown2ScrollDown = hotkey.new("", "down",
  -- when pressed
  function() scrollVertically(-3) end,
  -- when repeating / while held down
  function() scrollVertically(-1) end
):enable()
```

**Reply by skearya** [reply]
Script didnt work, for a very strange reason...?
I tried reinstalling hammerspoon because it seems like my install was corrupted, didn't help


2022-07-12 16:46:41: -- Lazy extension loading enabled
2022-07-12 16:46:41: -- Loading ~/.hammerspoon/init.lua
2022-07-12 16:46:41: *** ERROR: ...poon.app/Contents/Resources/extensions/hs/_coresetup.lua:662: module 'hs.eventtap.event' not found:
	no field package.preload['hs.eventtap.event']
	no file '/Users/bradley/.hammerspoon/hs/eventtap/event.lua'
	no file '/Users/bradley/.hammerspoon/hs/eventtap/event/init.lua'
	no file '/Users/bradley/.hammerspoon/Spoons/hs/eventtap/event.spoon/init.lua'
	no file '/opt/homebrew/share/lua/5.4/hs/eventtap/event.lua'
	no file '/opt/homebrew/share/lua/5.4/hs/eventtap/event/init.lua'
	no file '/opt/homebrew/lib/lua/5.4/hs/eventtap/event.lua'
	no file '/opt/homebrew/lib/lua/5.4/hs/eventtap/event/init.lua'
	no file '/usr/local/share/lua/5.4/hs/eventtap/event.lua'
	no file '/usr/local/share/lua/5.4/hs/eventtap/event/init.lua'
	no file '/usr/local/lib/lua/5.4/hs/eventtap/event.lua'
	no file '/usr/local/lib/lua/5.4/hs/eventtap/event/init.lua'
	no file './hs/eventtap/event.lua'
	no file './hs/eventtap/event/init.lua'
	no file '/private/var/folders/h_/b_tdh_cj6vxf97n5rkq0g2200000gn/T/AppTranslocation/648AC578-4AD2-4690-8BBA-E6C9E21FAC16/d/Hammerspoon.app/Contents/Resources/extensions/hs/eventtap/event.lua'
	no file '/private/var/folders/h_/b_tdh_cj6vxf97n5rkq0g2200000gn/T/AppTranslocation/648AC578-4AD2-4690-8BBA-E6C9E21FAC16/d/Hammerspoon.app/Contents/Resources/extensions/hs/eventtap/event/init.lua'
	no file '~/.local/share/hammerspoon/site/hs/eventtap/event.lua'
	no file '~/.local/share/hammerspoon/site/hs/eventtap/event/init.lua'
	no file '~/.local/share/hammerspoon/site/Spoons/hs/eventtap/event.spoon/init.lua'
	no file '/Users/bradley/.hammerspoon/hs/eventtap/event.dylib'
	no file '/Users/bradley/.hammerspoon/hs/eventtap/event.so'
	no file '/opt/homebrew/lib/lua/5.4/hs/eventtap/event.so'
	no file '/opt/homebrew/lib/lua/5.4/loadall.so'
	no file '/usr/local/lib/lua/5.4/hs/eventtap/event.so'
	no file '/usr/local/lib/lua/5.4/loadall.so'
	no file './hs/eventtap/event.so'
	no file '/private/var/folders/h_/b_tdh_cj6vxf97n5rkq0g2200000gn/T/AppTranslocation/648AC578-4AD2-4690-8BBA-E6C9E21FAC16/d/Hammerspoon.app/Contents/Frameworks/hs/eventtap/event.dylib'
	no file '~/.local/share/hammerspoon/site/lib/hs/eventtap/event.dylib'
	no file '~/.local/share/hammerspoon/site/lib/hs/eventtap/event.so'
	no file '/Users/bradley/.hammerspoon/hs.dylib'
	no file '/Users/bradley/.hammerspoon/hs.so'
	no file '/opt/homebrew/lib/lua/5.4/hs.so'
	no file '/opt/homebrew/lib/lua/5.4/loadall.so'
	no file '/usr/local/lib/lua/5.4/hs.so'
	no file '/usr/local/lib/lua/5.4/loadall.so'
	no file './hs.so'
	no file '/private/var/folders/h_/b_tdh_cj6vxf97n5rkq0g2200000gn/T/AppTranslocation/648AC578-4AD2-4690-8BBA-E6C9E21FAC16/d/Hammerspoon.app/Contents/Frameworks/hs.dylib'
	no file '~/.local/share/hammerspoon/site/lib/hs.dylib'
	no file '~/.local/share/hammerspoon/site/lib/hs.so'
stack traceback:
	[C]: in function 'rawrequire'
	...poon.app/Contents/Resources/extensions/hs/_coresetup.lua:662: in function 'require'
	/Users/bradley/.hammerspoon/init.lua:2: in main chunk
	[C]: in function 'xpcall'
	...poon.app/Contents/Resources/extensions/hs/_coresetup.lua:723: in function 'hs._coresetup.setup'
	(...tail calls...)

**Reply by MuhammedZakir** [reply]
Try changing second line to `local eventtap_event = hs.eventtap.event`.

**Reply by skearya** [reply]
thank you so much!
